distribution/test/itests/test-itests-kernel/src/test/java/org/codice/ddf/itests/kernel/ITSynchronizedInstaller.java

```diff
@@ -0,0 +1,347 @@
+import org.codice.ddf.test.ExampleMSFInstance;
+import org.codice.ddf.test.ExampleService;
+import org.codice.ddf.test.common.DependencyVersionResolver;
+import org.codice.ddf.test.common.annotations.BeforeExam;
+import org.codice.ddf.test.common.annotations.PaxExamRule;
+import org.codice.ddf.test.common.features.FeatureUtilities;
+import org.codice.ddf.test.common.features.TestUtilitiesFeatures;
+import org.junit.After;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.ops4j.pax.exam.Configuration;
+import org.ops4j.pax.exam.Option;
+import org.ops4j.pax.exam.junit.PaxExam;
+import org.ops4j.pax.exam.spi.reactors.ExamReactorStrategy;
+import org.ops4j.pax.exam.spi.reactors.PerClass;
+import org.osgi.framework.Bundle;
+import org.osgi.framework.BundleContext;
+import org.osgi.framework.InvalidSyntaxException;
+import org.osgi.framework.ServiceReference;
+import org.osgi.framework.ServiceRegistration;
+import org.osgi.service.cm.ConfigurationAdmin;
+
+@RunWith(PaxExam.class)
+@ExamReactorStrategy(PerClass.class)
+public class ITSynchronizedInstaller {
+
+  private static final String EXAMPLE_FEATURE = "example-feature";
+
+  private static final String EXAMPLE_BUNDLE_SYM_NAME = "example-bundle";
+
+  private static final String TEST_FEATURE_PATH = getTestResource("/test-features.xml");
+
+  @Inject private BundleContext bundleContext;
+
+  @Inject private SynchronizedInstaller syncInstaller;
+
+  @Inject private FeaturesService featuresService;
+
+  @Inject private ConfigurationAdmin configAdmin;
+
+  private static Bundle testBundle;
+
+  @Rule public PaxExamRule paxExamRule = new PaxExamRule(this);
+
+  @Configuration
+  public static Option[] examConfiguration() {
+    return options(
+        kernelDistributionOption(),
+        defaultVmOptions(),
+        defaultDebuggingOptions(),
+        defaultPortsOptions(),
+        defaultLogging(),
+        logLevelOption("org.codice.ddf.test", "TRACE"),
+        includeTestResources(),
+        addFeatureRepo(FeatureUtilities.toFeatureRepo(TEST_FEATURE_PATH)),
+        addBootFeature(TestUtilitiesFeatures.testCommon(), TestUtilitiesFeatures.awaitility()));
+  }
+
+  @BeforeExam
+  public void beforeExam() throws Exception {
+    startExampleBundle();
+    featuresService.installFeature(EXAMPLE_FEATURE);
+  }
+
+  public void startExampleBundle() throws Exception {
+    if (testBundle == null) {
+      testBundle =
+          bundleContext.installBundle(
+              maven()
+                  .groupId("ddf.test")
+                  .artifactId("example-bundle")
+                  .version(DependencyVersionResolver.resolver())
+                  .getURL());
+    }
+    testBundle.start();
+  }
+
+  @After
+  public void afterTest() throws Exception {
+    startExampleBundle();
+    featuresService.installFeature(EXAMPLE_FEATURE);
+    resetExampleMSProps();
+    deleteAllExampleMSFServices();
```

Comment:
If I can't reset these services and the tests continue executing, the test container is going to move into an unrecoverable state. Adding finally blocks here really won't do any good to prevent cascading failures. I wasn't unable to find a good way for pax exam to quit executing the remaining tests if an exception is thrown and shut the container down gracefully. I say we investigate this further if this impacts us in the future.